### ISSUE TITLE
docs: Add guidance for usage with Deno bundler

### ISSUE THREAD

**Issue by kamranayub** [excaliburjs/Excalibur#2062]
With [Deno](https://deno.land) there are a couple extra steps to consume Excalibur to bundle a game for the browser.

esm.sh should generate a bundle that works, tested locally.

```
deno info https://esm.sh/excalibur

type: JavaScript
dependencies: 180 unique (total 816KB)

https://esm.sh/excalibur (192B)
├── https://cdn.esm.sh/v54/excalibur@0.25.0/deno/excalibur.js (355.24KB)
```

Outputs a bundle:

```sh
deno bundle https://esm.sh/excalibur excalibur.bundle.js
```

### Using native in browser

Using as a native module works in Chrome without any extras:

```html
<script type="module">
  import ex from "./excalibur.bundle.js";

  const game = new ex.Engine();
  game.start();
</script>
```

### Using with bundler

More likely users will be writing a game and will be importing Excalibur. Deno uses URLs so we'll want to showcase pinning.

A custom `tsconfig.json` has to be used with `strict` turned off and there are a few DOM libs to add:

**tsconfig.json**

```json
{
  "compilerOptions": {
    "strict": false,
    "lib": ["dom", "dom.iterable", "dom.asynciterable", "deno.ns"]
  }
}
```

Then Excalibur can be imported from esh.sh:

**index.ts**

```ts
import { Engine } from "https://esm.sh/excalibur";

const game = new ex.Engine();
game.start();
``` 

And Deno will successfully bundle:

```sh
deno bundle index.ts game.bundle.js --config tsconfig.json
```

```
Check file:///E:/Development/Junk/deno-excalibur/index.ts
Bundle file:///E:/Development/Junk/deno-excalibur/index.ts
Emit "game.bundle.js" (678.91KB)
```

Thanks to @tarsupin for bringing this up.

_Originally posted by @kamranayub in https://github.com/excaliburjs/Excalibur/discussions/2060#discussioncomment-1480720_

**Comment by Return180bpm** [comment]
Would love to do a write-up of this!

**Comment by kamranayub** [comment]
Ah dang it, good catch, my code example is wrong 😅 Let me try again and see if I run into more errors. Usually what I found was I was missing libs to add, and the other error I saw was due to mismatches in the Deno default types that messed with DOM lib (I think it was crypto related).

**Comment by kamranayub** [comment]
I had a typo in my code above, I was using

```
import * as ex from "https://esm.sh/excalibur";

const game = new ex.Engine();
game.start();
```

The problem looks like a bundling issue, because at the very bottom of the bundle I see this:

```
var qo = Hn(hr()), Ko = Hn(hr()), { ex: Yo  } = qo;
Ko.default;
const game = new Engine();
game.start();
```

And if I fix it to be `qo.Engine` everything works; it looks like the Deno bundler is stripping the imported alias/namespace instead of replacing `ex` with the minified `qo` object 🤷‍♂️ 

I'll look at the ESM bundling we have setup to see if that could be an issue too.

**Comment by kamranayub** [comment]
@Return180bpm I just pushed a quick commit that adds:

```json
"module": "build/dist/index.js"
```

Which should be respected (by Skypack at least).

**Comment by kamranayub** [comment]
That is the bundle contents itself, it is generated from Skypack.

There's a PR that merged (or will merge) that should add the module field and an ESM bundle to the output, we can try that once it's available.

**Comment by kamranayub** [comment]
Thanks, those will be fixed in an upcoming PR I am working on that will upgrade our dependencies

**Comment by Return180bpm** [comment]
@kamranayub, @eonarheim: This was my first PR. Thank You for all your help!
